.github/workflows/node.js.yml

```diff
@@ -12,18 +12,40 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-latest #Can change the runtime environment according to needs
+    runs-on: ${{ matrix.os }}
 
     strategy:
       matrix:
         node-version: [14.x, 16.x]
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+        os: [ubuntu-latest, macOS-latest, windows-latest]
 
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      
+    - name: Cache node modules
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v2
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm run build
+    
+    - name: Install Node packages
+      run: npm install
```

Comment:
npm ci should run npm install right ?

Reply:
I wouldn't know, I used to follow `npm install` to install the node_modules.

Please tell me what to do?

Reply:
Use NPM CI 

It is used for installing on CI machines 
https://www.geeksforgeeks.org/difference-between-npm-i-and-npm-ci-in-node-js/

Reply:
Thanks for the information